teuthology/worker.py

```diff
@@ -191,8 +191,6 @@ def prep_job(job_config, log_file_path, archive_dir):
 def run_job(job_config, teuth_bin_path, archive_dir, verbose):
     safe_archive = safepath.munge(job_config['name'])
     if job_config.get('first_in_suite') or job_config.get('last_in_suite'):
-        if teuth_config.results_server:
-            report.try_delete_jobs(job_config['name'], job_config['job_id'])
```

Comment:
Seems like we still need this to support older users, since they may not be able to update their teuthology sandboxes to the latests code corresponding the job will continue added by some users.
I only see that we can add try-catch block to save worker from dying and maybe add couple of tries with pause to handle situations which we have downstream, when paddles is got the job to be add in some thread awaiting it's own turn and worker already picked the the LIS and FIS jobs and started to processing them trying to delete job which is not created yet.

Reply:
Corresponding PR #1524